packages/plugin/src/__tests__/compile.test.ts

```diff
@@ -1,5 +1,48 @@
-import { test } from 'vitest';
+import { expect, test } from "vitest";
+import { compile } from "../compile.js";
+import {
+  MOCK_EXPORTS_WITH_PROMISE,
+  MOCK_JSON_SAFE_EXPORTS,
+  MOCK_JSON_UNSAFE_EXPORTS,
+} from "../__mocks__/compile.mock.js";
 
-test.todo('generates default export');
-test.todo('generates named exports');
-test.todo('generates resolved promises');
+test("generates default export", async () => {
+  const code = await compile(MOCK_JSON_SAFE_EXPORTS);
```

Comment:
Rather than using constants, I would use inline values that only contain the input that generates the type of output being tested. So in this case, something like `{ default: 2 }` would be fine.